File: src/main/java/com/owiseman/dataapi/plugins/sdk/PluginConfig.java
package com.owiseman.dataapi.plugins.sdk;

import java.util.HashMap;
import java.util.Map;

/**
 * 插件配置类
 */
public class PluginConfig {
    private String serverHost;
    private int serverPort;
    private String pluginId;
    private String pluginName;
    private String pluginVersion;
    private String pluginType;
    private Map<String, String> additionalConfig = new HashMap<>();

    public String getServerHost() {
        return serverHost;
    }

    public void setServerHost(String serverHost) {
        this.serverHost = serverHost;
    }

    public int getServerPort() {
        return serverPort;
    }

    public void setServerPort(int serverPort) {
        this.serverPort = serverPort;
    }

    public String getPluginId() {
        return pluginId;
    }

    public void setPluginId(String pluginId) {
        this.pluginId = pluginId;
    }

    public String getPluginName() {
        return pluginName;
    }

    public void setPluginName(String pluginName) {
        this.pluginName = pluginName;
    }

    public String getPluginVersion() {
        return pluginVersion;
    }

    public void setPluginVersion(String pluginVersion) {
        this.pluginVersion = pluginVersion;
    }

    public String getPluginType() {
        return pluginType;
    }

    public void setPluginType(String pluginType) {
        this.pluginType = pluginType;
    }

    public Map<String, String> getAdditionalConfig() {
        return additionalConfig;
    }

    public void setAdditionalConfig(Map<String, String> additionalConfig) {
        this.additionalConfig = additionalConfig;
    }

    public void addConfig(String key, String value) {
        this.additionalConfig.put(key, value);
    }

    public String getConfig(String key) {
        return this.additionalConfig.get(key);
    }
}

File: cpp/plugin_config.cpp
#include "plugin_config.h"

PluginConfig::PluginConfig() : serverPort(0) {
}

std::string PluginConfig::getServerHost() const {
    return serverHost;
}

void PluginConfig::setServerHost(const std::string& serverHost) {
    this->serverHost = serverHost;
}

int PluginConfig::getServerPort() const {
    return serverPort;
}

void PluginConfig::setServerPort(int serverPort) {
    this->serverPort = serverPort;
}

std::string PluginConfig::getPluginId() const {
    return pluginId;
}

void PluginConfig::setPluginId(const std::string& pluginId) {
    this->pluginId = pluginId;
}

std::string PluginConfig::getPluginName() const {
    return pluginName;
}

void PluginConfig::setPluginName(const std::string& pluginName) {
    this->pluginName = pluginName;
}

std::string PluginConfig::getPluginVersion() const {
    return pluginVersion;
}

void PluginConfig::setPluginVersion(const std::string& pluginVersion) {
    this->pluginVersion = pluginVersion;
}

std::string PluginConfig::getPluginType() const {
    return pluginType;
}

void PluginConfig::setPluginType(const std::string& pluginType) {
    this->pluginType = pluginType;
}

const std::map<std::string, std::string>& PluginConfig::getAdditionalConfig() const {
    return additionalConfig;
}

void PluginConfig::setAdditionalConfig(const std::map<std::string, std::string>& additionalConfig) {
    this->additionalConfig = additionalConfig;
}

void PluginConfig::addConfig(const std::string& key, const std::string& value) {
    this->additionalConfig[key] = value;
}

std::string PluginConfig::getConfig(const std::string& key) const {
    auto it = additionalConfig.find(key);
    if (it != additionalConfig.end()) {
        return it->second;
    }
    return "";
}